packages/fields/src/fields/template/template-edit.tsx

```diff
@@ -39,52 +39,49 @@ export const TemplateEdit = ( {
 		typeof data.id === 'number' ? data.id : parseInt( data.id, 10 );
 	const slug = data.slug;
 
-	const { availableTemplates, templates } = useSelect(
+	const { templates, allowSwitchingTemplate } = useSelect(
 		( select ) => {
-			const allTemplates =
-				select( coreStore ).getEntityRecords< WpTemplate >(
-					'postType',
-					'wp_template',
-					{
-						per_page: -1,
-						post_type: postType,
-					}
-				) ?? [];
+			const allTemplates = select(
+				coreStore
+			).getEntityRecords< WpTemplate >( 'postType', 'wp_template', {
+				per_page: -1,
+				post_type: postType,
+			} );
 
 			const { getHomePage, getPostsPageId } = unlock(
 				select( coreStore )
 			);
 
-			const isPostsPage = getPostsPageId() === +postId;
+			const isPostsPage = +getPostsPageId() === postId;
 			const isFrontPage =
-				postType === 'page' && getHomePage()?.postId === +postId;
-
-			const allowSwitchingTemplate = ! isPostsPage && ! isFrontPage;
+				postType === 'page' && +getHomePage()?.postId === postId;
```

Comment:
Small fix, the `getPostsPageId()` and `getHomePage()?.postId` actually return strings, so `allowSwitchingTemplate` always returned true.

Reply:
Should we make `getHomePage` return the right type?

Reply:
Meaning an integer? It seems like in some of the other logic you were already depending on it being a string: https://github.com/WordPress/gutenberg/blob/4867dfc6bb9286865ae19e332b06a29b35ce7ab1/packages/core-data/src/private-selectors.ts#L188
Similar scenario with the `getPostsPageId` that explicitly converts it to a string: https://github.com/WordPress/gutenberg/blob/4867dfc6bb9286865ae19e332b06a29b35ce7ab1/packages/core-data/src/private-selectors.ts#L115

Instead of converting it to an integer above, maybe I should just keep the `postId` as a string?

Reply:
If I'm not wrong, the "posts" REST API returns post Ids as string, except for templates and template Parts. So IMO, we should stay consistent with that. When the post type is "template" or "template part" use string ids, otherwise integer ids.

There might be some type casting we need to do at the route levels, because urls always give strings. 

WDYT?

Reply:
> If I'm not wrong, the "posts" REST API returns post Ids as string

@youknowriad, do you mean integers? REST API should return DB ID values as ints.

Reply:
yes sorry :)

Reply:
Yeah I agree we should stay consistent, and happy to stick with `When the post type is "template" or "template part" use string ids, otherwise integer ids.`.
Would it help to have a `sanitize` function for this that includes the `postType` and uses the above logic?

Things may get a bit confusing in functions like  `getHomePage` ( https://github.com/WordPress/gutenberg/blob/4867dfc6bb9286865ae19e332b06a29b35ce7ab1/packages/core-data/src/private-selectors.ts#L144-L156 ) where it returns either a `page` or `wp_template` so the `postId` will likewise be either a string or integer. Which in a way I see why you piped it through `normalizePageId` to convert it to a string.

Reply:
I updated the above function to use integers when dealing with pages or posts in this commit: https://github.com/WordPress/gutenberg/pull/67872/commits/ba6868a3a10c615c20c1bc53f9c53f74db49f024
I also converted the list of ids coming from the router to integers, as part of this commit: https://github.com/WordPress/gutenberg/pull/67872/commits/9c55bef36e3190e74e5864f0b3a56989b12f7509